### DISCUSSION TITLE
Concerning deploying and running transactions

### DISCUSSION THREAD

**Discussion by Taffy91** [Cyfrin/security-and-auditing-full-course-s23#30] [Q&A]
My first time here.  I'm currently doing the security and auditing contract course and I am curious to know how to move forward having encountered these problems. In lesson 10 of the Section 1: Review, Patrick showed that when ready to deploy and run the transaction early on in the video, 3 minutes into the video that the 'combine strings' option is missing from the deploy and run transactions tab. Also, the Javascript VM option is missing from 'environment'. Someone on another post advised to use 'Remix VM (London)' instead as that option was present and it would work perfectly. But upon using Remix VM London, right above the 'Deploy' button, in very small text, was written, 'evm version: shanghai'. Upon placing my cursor upon that small text above the Deploy button a message appeared, which only comes when 'evm version: shanghai' is highlighted by the cursor. The message reads, 'Please make sure that these current network is compatible with this evm version: shanghai. Otherwise any deployment will fail.  I'm only asking this question because on another post it was specifically advised to use Remix VM (London) and not Remix VM (Shanghai). Is it ok to use Shanghai? What should I use? What should I do? Thanks in advance all.

**Comment by kwamentw** [comment]
i think you can use any so far as it is a "remix VM" you are okay

**Reply by Taffy91** [reply]
Thank you. I just took a chance a chose Remix VM Shanghai and it worked actually. But thanks a lot for your response.

**Reply by cromewar** [reply]
nice question, I wouldn't pay that much attention to the version on REMIX on most of scenarios it would work anyway.
